### PR TITLE
Allow linking test titles to specs.

### PR DESCRIPTION
**Pull request by BigBlueHat**
Rather than just having this at the comment level, we can move these into the test data itself.

The links to add would need to be added to the test data in a similar way as `test.cell` stuff is added here:
https://github.com/w3c/vc-data-model-2.0-test-suite/blob/main/tests/10-vcdm2.js#L40-L43

Not sure that's ideal...but it does work.

I can provide readme updates, if y'all think this approach is worth doing.

The easy way to add the data to a test is just...
```js
it('test', async function() {
  this.test.link = 'https://example.com/';
});
```

It's a little distracting...but works.

### CODE REVIEW

File: test/multiple-matrix.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
               "_testId": "urn:uuid:7d556c3e-3a4a-48da-bb11-a357c4ddb7f9",
               "_events": {},
               "_eventsCount": 1,
+              "link": "https://example.com/",
```
Comment (aljones15) [inline test/multiple-matrix.json:31]: This leave link hanging out with the mocha test properties, I think it's better to move it inside of cell so `cell.link` then the link can be used with the rowId or columnId in fact we should even row something like rowLink columnLink or something like that in the future, but just link for right now seems fine. Has this been run with an actual test suite?

**Review by aljones15 — APPROVED** [review]
Approved provided it can be tested with an actual suite before release!

File: test/multiple-matrix.json (hunk 1 above)
Comment (BigBlueHat) [reply test/multiple-matrix.json:31]: It's not per-cell, but per-test... Though, in thinking of this more, there really should be a "metadata" space we populate similar to how https://github.com/mongodb-js/metamocha adds it's data.

**Comment by BigBlueHat** [comment]
> p.s. also test it with a suite with no links to ensure it works with reports with and with out links, but I think the modified test data already covers that.

Yeah, I've been testing it with VCDM tests, but I think this still needs work. I'm not happy with overloading the test title parameter...it should really be more of an augmentation.

**Comment by BigBlueHat** [comment]
Did more local testing and this is good to go--and I plan to use it soon on the VCDMv2 test suite, so merging.

**Pull request merged**